docs/standards/v1/STANDARDS.md

```diff
@@ -0,0 +1,322 @@
+# Hack Oregon Data APIs v1 Style Guide
+
+## Overview
+
+The intent of this document is to provide some guidance for API developers when building APIs under Hack Oregon Projects. Our aim is to follow general RESTful API best practices, and work with solutions/standards that are built into our technology stack.
+
+We aim to also provide the highest level of interoperability between our APIs. By adopting standards between teams, we can better support users connecting to multiple APIs.
+
+### API Technologies
+* **HTTPS:** All API requests must be made over HTTPS.
+* **JSON:** All API request and response bodies will be JSON.
+
+## Requests
+
+Requests and endpoints should follow a common REST-based structure.
+
+### URL Structure
+
+* URLs will use the Hack Oregon team name as the root:  
+  Pattern: `:team_name/v1/:resource name`   
+  Example: `transportation_systems/v1/stops`  
+
+
+* URLs will be prefixed by the major version number:  
+  Pattern: `/v1/:resource name`  
+  Example: `/v1/stops`  
+
+* Collections of resources are referenced by their resource name (plural):  
+  Pattern: `/v1/:resource name`  
+  Example: `/v1/stops`
+
+* Individual resources are referenced by their resource name (plural) followed by the guid:  
+  Pattern: `/v1/:resource name/:guid`  
+  Example: `/v1/stops/25fe21b8-8de2-40d0-93b0-c819101d1a11`
+
+
+### GET
+
+Request a single resource or collection of Resources
+
+* GET requests may include query parameters
+* GET requests must NOT include a request body
+
+#### Single Resource
+
+```
+GET /v1/stops/25fe21b8-8de2-40d0-93b0-c819101d1a11
+```
+
+##### Responses (Resource)
+
+|Scenario|Code|Body|
+|---|---|---|
+| Valid Resource/Request | 200 | Resource |
+| Bad Request | 400 | Error |
+| Not Found | 404 | Error |
+| Internal Server Error | 500 | Error |
+
+#### Collection
+
+```
+GET /v1/stops/
+```
+
+##### Responses (Resource)
+
+|Scenario|Code|Body|
+|---|---|---|
+| Valid Resource/Request | 200 | List of Resources (All or Paginated) |
+| Valid Resource, No Results | 200 | Empty Body|
+| Bad Request | 400 | Error |
+| Internal Server Error | 500 | Error |
+
+#### Recommended Technologies to Support
+
+* Django Restframework Viewsets and Mixins should generally handle this behavior. If you are writing custom views, use built-in status codes.
+
+* Users will have to correctly provide routing
```

Comment:
Specify which users (API developers? API consumers?)

Reply:
clarified:

"* API developers will have to correctly define routing"

https://github.com/hackoregon/2019-backend-cookiecutter-django/commit/45f51e02c73641704f2249d58bd21f813d56324c